### DISCUSSION TITLE
patching shealth with shealth_custom_cert.patch

### DISCUSSION THREAD

**Discussion by ShaDisNX255** [adil192/SamsungAppsPatcher#9] [General]
I know you're officially stopped developing for this, but if you would be so kind to help me out here
It looks like I'm running into this issue

```
patching shealth with shealth_custom_cert.patch
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- smali_classes9/com/samsung/android/app/shealth/util/SignatureChecker.smali
|+++ smali_classes9/com/samsung/android/app/shealth/util/SignatureChecker.smali
```

If you could guide me on how to bypass or fix this I would appreciate it. I'm still kind of learning how your script works, I'm not that knowledgeable

Thanks

**Comment by ShaDisNX255** [comment]
Update: Ok, it moved location. It's in smali_classes 2.
I updated shealth_custom_cert.patch with this information but now I'm getting this problem
```

Hunk #1 FAILED at 24.
Hunk #2 FAILED at 101.
Hunk #3 succeeded at 105 with fuzz 2 (offset -28 lines).
2 out of 3 hunks FAILED -- saving rejects to file smali_classes2/com/samsung/android/app/shealth/util/SignatureChecker.smali.rej
```

It stops here.

**Comment by adil192** [comment]
Every few updates the patches need to be updated because the files in the original APK have rejigged themselves.
The patch you're using is for the [6.22.0.069](https://www.apkmirror.com/apk/samsung-electronics-co-ltd/s-health/s-health-6-22-0-069-release/) update (2 June) but you're likely using the latest APK [6.21.3.001](https://www.apkmirror.com/apk/samsung-electronics-co-ltd/s-health/s-health-6-21-3-001-release/) (27 Jun).

The answer in this case is to use the older 6.21 patch here: [shealth_custom_cert.patch](https://github.com/adil192/SamsungAppsPatcher/blob/11e833b7004902b679139d7d2b9e3b6fd551464f/patches/shealth_custom_cert.patch).

This usually wouldn't happen but Samsung released a 6.21 update \~after\~ a 6.22 update for unknown reasons which requires the older patch.

**Reply by ShaDisNX255** [reply]
Wow that is so weird, I've seen this a few times from Samsung lol. Thank you for the information.

**Reply by faakarna** [reply]
> Wow that is so weird, I've seen this a few times from Samsung lol. Thank you for the information.

Did you Seem to fix the issue. Really cant Seem to patch the Samsung wearable apps my self. Would like to know if you managed to patch the newest apks? And if you are willing to share the patched apks? Simply cant hide.the root for the wearable.apk, so my watch are useless om my A52 5g. So annoying.

**Reply by ShaDisNX255** [reply]
> > Wow that is so weird, I've seen this a few times from Samsung lol. Thank you for the information.
> 
> Did you Seem to fix the issue. Really cant Seem to patch the Samsung wearable apps my self. Would like to know if you managed to patch the newest apks? And if you are willing to share the patched apks? Simply cant hide.the root for the wearable.apk, so my watch are useless om my A52 5g. So annoying.

I haven't tried. I only use 3 apps which are Health, Wear and Buds Live Manager. I've only received an update notification for Health and Wear so those are the first I'll attempt to patch. I've forked this repo so I will create a pull request if I manage to get those working at the very least

**Reply by adil192** [reply]
> I haven't tried. I only use 3 apps which are Health, Wear and Buds Live Manager. I've only received an update notification for Health and Wear so those are the first I'll attempt to patch. I've forked this repo so I will create a pull request if I manage to get those working at the very least

@ShaDisNX255 Seems like Samsung have started to obfuscate their code with the latest shealth update: I've updated the shealth patches (untested of course) on this repo if you'd like to try again

**Reply by adil192** [reply]
> Simply cant hide.the root for the wearable.apk, so my watch are useless om my A52 5g. So annoying.

@faakarna Can you share any more about this? Is the Wearable app detecting root?

**Reply by faakarna** [reply]
Sorry. I wasnt precisely enough. 
Your patched apks are working great, but now wearable apk wont let med use your latest patched wearable apk, because IT wants me to update. 

Its only the official Samsung wearable apps, that detects root, nomatter how Many magisk fixes, i have tried. 
But i Will look further into patching the apks. Thanks for your effort, and some great working patches. Appreciate that.

**Reply by ShaDisNX255** [reply]
> > I haven't tried. I only use 3 apps which are Health, Wear and Buds Live Manager. I've only received an update notification for Health and Wear so those are the first I'll attempt to patch. I've forked this repo so I will create a pull request if I manage to get those working at the very least
> 
> @ShaDisNX255 Seems like Samsung have started to obfuscate their code with the latest shealth update: I've updated the shealth patches (untested of course) on this repo if you'd like to try again

I'll try to patch it tonight. I rarely have free time on weekdays but I want to give you an answer as soon as I can. Thanks for helping out, really appreciate it.

**Reply by adil192** [reply]
> now wearable apk wont let med use your latest patched wearable apk, because it wants me to update.  

New wearable update on [the website](https://adil.hanney.org/SamsungAppsPatcher)!

**Reply by faakarna** [reply]
Thank you very much for updated wearable apks.  Now wearable are usable again. Tried to download new version of shealth om your website. But the link seems Dead. 🤔

**Reply by adil192** [reply]
> Thank you very much for updated wearable apks. Now wearable are usable again. Tried to download new version of shealth om your website. But the link seems Dead. thinking

For some reason it didn't upload °~°, should be fixed now

**Comment by faakarna** [comment]
Sounds great. Spil love to know, og you get it working. Been trying om
Windows, with No luck. Not very experienced in Linux.
Been using health, watch3plugin and wearable. Wearable are pushing for
update, so cant use the app anymore, without patching new version.
Will follow, in github, if you managed to somehow patch the apks. Thanks
for the answer.



tor. 11. aug. 2022 14.44 skrev ShaDisNX255 ***@***.***>:
